### PR TITLE
feat(be+fe): export data catalog

### PR DESCRIPTION
**Pull request by 530051970**
**What is the current behavior?

Please describe the current behavior of the code that you are trying to modify. This could include any issues or limitations that you are addressing.

**What is the updated behavior?

Please describe the changes you are making to the code's behavior. This could include new features or enhancements that you are adding, as well as how they differ from the current behavior.

**Checklist

Please check off the following items before submitting your pull request:

- [x] I have tested my changes.
- [x] No sensitive information is included.
- [x] I've reviewed my changes to ensure they won't cause any new issues that could affect the stability or performance of the codebase.

Please check off the following items if this changes include API modefications:

- [x] I confirm this changes has been test with [authorization validation steps](https://github.com/awslabs/sensitive-data-protection-on-aws/blob/main/CONTRIBUTING.md#contributing-via-pull-requests), it won't break the authorization token verification mechanism.
- [x] I confirm this changes won't cause security issues.

### CODE REVIEW

**Review by yanbasic — APPROVED** [review]
LGTM

File: source/constructs/api/catalog/service.py [hunk 1]
```diff
@@ -962,3 +967,115 @@ def fill_catalog_labels(catalogs):
         labels_str = [{"id": label.id, "label_name": label.label_name} for label in labels]
         catalog.labels = labels_str
     return catalogs
+
+def get_catalog_export_url(fileType: str, timeStr: str) -> str:
+    run_result = crud.get_export_catalog_data()
+    logger.info(f"&&&&&&&&&&&&&&&&&&&&&& run_result size is {len(run_result)}")
```
Comment (yanbasic) [inline source/constructs/api/catalog/service.py:973]: ?

File: source/constructs/api/catalog/service.py [hunk 2]
```diff
@@ -962,3 +967,115 @@ def fill_catalog_labels(catalogs):
...
+    tmp_filename = f"/tmp/catalog_{timeStr}.zip"
+    report_file = f"report/catalog_{timeStr}.zip"
+    # key_name = f"report/{filename}" 
+    for item in all_labels:
+        all_labels_dict[item.id] = item.label_name
+    filtered_records = filter_records(run_result, all_labels_dict)
+    column_header = {const.EXPORT_S3_MARK_STR: const.EXPORT_FILE_S3_COLUMNS, const.EXPORT_RDS_MARK_STR: const.EXPORT_FILE_RDS_COLUMNS}
+    gen_zip_file(column_header, filtered_records, tmp_filename, fileType)
+    stats = os.stat(tmp_filename)
+    logger.info(f"############ tmp_filename size is {stats.st_size}")
```
Comment (yanbasic) [inline source/constructs/api/catalog/service.py:986]: ?

File: source/constructs/api/catalog/service.py [hunk 3]
```diff
@@ -962,3 +967,115 @@ def fill_catalog_labels(catalogs):
...
+    s3_client = boto3.client('s3')
+    if stats.st_size < 6 * 1024 * 1024:
+        s3_client.upload_file(tmp_filename, project_bucket_name, report_file)
+    else:
+        logger.info("%%%%%%%%%%%%%%%%%%%%%%%%%concurrent_upload")
```
Comment (yanbasic) [inline source/constructs/api/catalog/service.py:991]: please remove

**Review by yanbasic — CHANGES_REQUESTED** [review]

File: source/constructs/api/catalog/service.py (hunk 1 above)
Comment (530051970) [reply source/constructs/api/catalog/service.py]: resloved.

File: source/constructs/api/catalog/service.py (hunk 2 above)
Comment (530051970) [reply source/constructs/api/catalog/service.py]: resloved

File: source/constructs/api/catalog/service.py (hunk 3 above)
Comment (530051970) [reply source/constructs/api/catalog/service.py]: resloved.

**Comment by 530051970** [comment]
due to new commit, close this pr.

**Pull request closed without merging**